### ISSUE TITLE
TDM not seeing drop channels for Paladins

### ISSUE THREAD

**Issue by smallgarage** [DevilXD/TwitchDropsMiner#233]
Hi, thanks so much for this great software. Have been using it for some time now, and it keeps getting better =)

I'm tried using the latest build from source (8189f01) since I noticed this begin happening while using another recent build. The miner isn't seeing any active drop channel for Paladins while there are definitely more than 1 active. Manually watching ttv/PaladinsGame in a browser does increase drop progress, and I see "Earned points for watching" messages in TDM related to that, along with "No available channels to watch. Waiting for an ONLINE channel...".

Recently I added a few more games to the mining list, but nothing else changed on my end.

Any ideas? Seems to only be affecting this game. It's also the only uncompleted campaign showing in the Inventory section.

**Comment by DovydasTEDS** [comment]
I have had the same, but with other vgames (I think all?)

**Comment by DevilXD** [comment]
This kinda makes no sense what so ever, but I tried it anyway, and...

https://github.com/DevilXD/TwitchDropsMiner/commit/a7c3925e53f211fc032dde169220b01b1e55d89a removes the "drops enabled" tag requirement from the channel search. The resulting list of channels appears to contain channels with the expected tag anyway.

Please try out the latest dev build to see if that works better.

**Comment by smallgarage** [comment]
It definitely wasn't making much sense. I'll try again when the new drop campaign starts later today and let you know. Thanks for the quick response & build!

Btw, I did notice that at least 1 of the active channels was using the tag "Drops Enabled" or "DropsEnabled" while this was happening.

**Comment by DevilXD** [comment]
The tags you're seeing are "free form" tags. They have no actual tag ID associated to them, only the text string itself. They're completely unrelated of the actual "drops enabled" setting a streamer can set on their dashboard - one can enable drops but don't add any tags, or add a tag and don't enable the drops. Because of this, they cannot be relied upon, at all. Even if they could be considered as "a hint" of the tag being enabled, them being completely free form means the tag can come in many different languages, spellings, casings, and versions overall, making any system of searching for and filtering with the tag - impossible.

In other words, you can ignore whatever tags you see under the videos, the miner cannot use them in any useful way.

**Comment by smallgarage** [comment]
Thanks for the explanation - that makes sense. It's unfortunate Twitch doesn't have a better system for this.

Happy to report that the new commit resolved the issue as well. Thank you again!

**Comment by DevilXD** [comment]
> It's unfortunate Twitch doesn't have a better system for this.

The point is, they *had* a better system for this in the past, but decided to scrap it and replace it with those free form tags. This "process of scrapping" caused #189 and #212 to appear, both of which are now temp-fixed by ignoring the tag requirement when considering channels as mining candidates, and now fetching channels without the requirement of the tag being present.

In any way, I'm glad it's fixed for you now 🙂

**Comment by smallgarage** [comment]
I thought so! Interesting. And as of now there's no way for the API (or apparently the regular web users) to know whether the streamer has actually flipped the "drops on" toggle, other than watching the channel and seeing if drop progress happens? What a pain to develop workarounds for, aside from the regular client auth woes.

If that's all true I wonder what their motivations for making this change were. Perhaps complexities of certain drops only being for certain channels - it seems the miner is already somehow aware of this case, but maybe that's available from the drop campaign info.

Overall it seems to me that the system of drops being turned on to be the only way to have the visible "Drops Enabled" tag and a corresponding API value was pretty straightforward logic. Even if people could add custom tags like "Drops Here", etc. people would know to look for (and hopefully be able to search for) streams with the specific tag on, and could easily handle cases where the channel was online but not playing the
